### PR TITLE
ddl: Add prometheus metrics

### PR DESCRIPTION
**Pull request by zimulala**

### CODE REVIEW

File: ddl/column.go [hunk 1]
```diff
@@ -307,21 +310,25 @@ func (d *ddl) backfillColumn(t table.Table, columnInfo *model.ColumnInfo, reorgI
...
+		sub := time.Since(startTS).Seconds()
 		err = d.backfillColumnData(t, columnInfo, handles, reorgInfo)
 		if err != nil {
+			log.Warnf("[ddl] added column for %v rows failed, take time %v", count, sub)
 			return errors.Trace(err)
 		}
 
-		count += len(handles)
-		log.Infof("[ddl] added column for %v rows", count)
+		batchHandleDataHistogram.WithLabelValues(batchAddCol, strconv.Itoa(len(handles))).Observe(sub)
```
Comment (siddontang) [inline ddl/column.go]: seem no need to use len(handles) as label value.

File: ddl/metrcs.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import "github.com/prometheus/client_golang/prometheus"
+
+var (
+	jobsGauge = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "tidb",
+			Subsystem: "ddl",
+			Name:      "waiting_jobs",
+			Help:      "Gauge of jobs.",
+		}, []string{"job_type", "action_type"})
```
Comment (siddontang) [inline ddl/metrcs.go]: type, action

**Comment by zimulala** [comment]
PTAL @siddontang

File: ddl/metrcs.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
+// Copyright 2016 PingCAP, Inc.
```
Comment (hanfei1991) [inline ddl/metrcs.go]: I think the filename should be metrics

Comment (hanfei1991) [inline ddl/metrcs.go]: I think the filename should be "metrics"

**Comment by shenli** [comment]
LGTM

**Comment by shenli** [comment]
@siddontang @hanfei1991 PTAL

**Comment by siddontang** [comment]
LGTM

**Pull request merged**